### ISSUE TITLE
[Bug] Bottlerocket nodes fail to join EKS cluster for 1.26

### ISSUE THREAD

**Issue by Himangini** [weaveworks/eksctl#6544]
This issue is for community visibility for 1.26 support. 

Our integration tests have been failing constantly and consistently ever since we merged the [support for 1.26](https://github.com/weaveworks/eksctl/pull/6502) work. This is released in the latest RC https://github.com/weaveworks/eksctl/releases/tag/v0.138.0-rc.0

Following is the account for the failing integration test and errors 
```
[FAIL] (Integration) Create Managed Nodegroups cluster with 1 al2 managed nodegroup and upgrading a nodegroup [It] should upgrade to the next Kubernetes version
---
Error: error updating nodegroup stack: waiter state transitioned to Failure

2023-04-20 01:51:34 [ℹ]  CloudFormation.events/eksctl-it-managed-unique-hideout-1681949755-nodegroup-bottlerocket-1[4] = {TimeStamp:2023-04-20 01:46:06.462 +0000 UTC LogicalID:eksctl-it-managed-unique-hideout-1681949755-nodegroup-bottlerocket-1 ResourceStatus:UPDATE_ROLLBACK_IN_PROGRESS ResourceStatusReason:The following resource(s) failed to update: [ManagedNodeGroup]. }
[0]   2023-04-20 01:51:34 [ℹ]  CloudFormation.events/eksctl-it-managed-unique-hideout-1681949755-nodegroup-bottlerocket-1[5] = {TimeStamp:2023-04-20 01:46:05.888 +0000 UTC LogicalID:ManagedNodeGroup ResourceStatus:UPDATE_FAILED ResourceStatusReason:Resource handler returned message: "[ErrorDetail(ErrorCode=NodeCreationFailure, ErrorMessage=Couldn't proceed with upgrade process as new nodes are not joining node group bottlerocket-1, ResourceIds=[])] (Service: null, Status Code: 0, Request ID: null)" (RequestToken: ac69a98d-13c5-8ca9-e841-d9af01e1d3df, HandlerErrorCode: GeneralServiceException)}
```

Seems to be related to :
https://github.com/bottlerocket-os/bottlerocket/issues/3013

We will be graduating the current RC when we have resolved this issue and we are currently blocked upstream.

I ran the suite locally on another testing account and noticed the same failure so looks like https://github.com/bottlerocket-os/bottlerocket/issues/3013, https://github.com/bottlerocket-os/bottlerocket/issues/3028 which are now marked as closed, hasn't resolved our situation.

Logs from local run
```
2023-04-20 13:15:32 [ℹ]  waiting for CloudFormation stack "eksctl-hi-it-managed-floral-monster-1681986666-nodegroup-bottlerocket-1"
  2023-04-20 13:16:32 [ℹ]  waiting for CloudFormation stack "eksctl-hi-it-managed-floral-monster-1681986666-nodegroup-bottlerocket-1"
  2023-04-20 13:17:59 [ℹ]  waiting for CloudFormation stack "eksctl-hi-it-managed-floral-monster-1681986666-nodegroup-bottlerocket-1"
  Error: error updating nodegroup stack: waiter state transitioned to Failure
  [FAILED] Expected '../../../eksctl "--region" "us-west-2" "upgrade" "nodegroup" "--verbose" "4" "--cluster" "hi-it-managed-floral-monster-1681986666" "--name" "bottlerocket-1" "--kubernetes-version" "1.26" "--timeout=60m"' to succeed, got return code 1
  In [It] at: /Users/himangini/Documents/GitHub/eksctl/integration/tests/managed/managed_nodegroup_test.go:626 @ 04/20/23 13:17:59.245
  < Exit [It] should upgrade to the next Kubernetes version - /Users/himangini/Documents/GitHub/eksctl/integration/tests/managed/managed_nodegroup_test.go:584 @ 04/20/23 13:17:59.245 (1h3m36.102s)

---
 2023-04-20 13:18:03 [ℹ]  CloudFormation.events/eksctl-hi-it-managed-floral-monster-1681986666-nodegroup-bottlerocket-1[4] = {TimeStamp:2023-04-20 12:11:40.659 +0000 UTC LogicalID:eksctl-hi-it-managed-floral-monster-1681986666-nodegroup-bottlerocket-1 ResourceStatus:UPDATE_ROLLBACK_IN_PROGRESS ResourceStatusReason:The following resource(s) failed to update: [ManagedNodeGroup]. }
  2023-04-20 13:18:03 [ℹ]  CloudFormation.events/eksctl-hi-it-managed-floral-monster-1681986666-nodegroup-bottlerocket-1[5] = {TimeStamp:2023-04-20 12:11:39.766 +0000 UTC LogicalID:ManagedNodeGroup ResourceStatus:UPDATE_FAILED ResourceStatusReason:Resource handler returned message: "[ErrorDetail(ErrorCode=NodeCreationFailure, ErrorMessage=Couldn't proceed with upgrade process as new nodes are not joining node group bottlerocket-1, ResourceIds=[])] (Service: null, Status Code: 0, Request ID: null)" (RequestToken: 3aaff7c1-4b2b-981f-28aa-34d34804c9d6, HandlerErrorCode: GeneralServiceException)}
  2023-04-20 13:18:03 [ℹ]  CloudFormation.events/eksctl-hi-it-managed-floral-monster-1681986666-nodegroup-bottlerocket-1[6] = {TimeStamp:2023-04-20 11:42:48.584 +0000 UTC LogicalID:ManagedNodeGroup ResourceStatus:UPDATE_IN_PROGRESS ResourceStatusReason:}
```

**Comment by yeazelm** [comment]
We are working on getting a new release out that should fix this but I noticed we never confirmed what specific issue caused the nodes from joining the cluster. @Himangini did you ever get any logs from the nodes or confirm you had one of the configurations we noticed were causing issues (namely custom DNS names, Resource Based Naming, or multiple DNS names in your DHCP options)? I just wanted to make sure we have captured why the nodes weren't joining so we are confident that our next release will unblock these tests for you.
